src/sudoku-checker.ts

```diff
@@ -1,3 +1,26 @@
+import { Sudoku } from "./interfaces";
+import { readSudokuFile } from "./utils";
+
 export class SudokuChecker {
-    public check() {}
+
+    private sudoku: Sudoku;
+
+    constructor(path: string) {
+        this.sudoku = readSudokuFile(path);
+    }
+
+    public check(): boolean {
+        return this.checkElement(this.sudoku.lines)
+        && this.checkElement(this.sudoku.columns)
+        && this.checkElement(this.sudoku.squares);
+    }
+
+    private checkElement(element: number[][]): boolean {
+        let verify = true;
+        element.forEach(array => {
+            verify = (new Set(array)).size === array.length;
```

Comment:
I don't know the `Set` object type, can you explain me a little bit what is it for?